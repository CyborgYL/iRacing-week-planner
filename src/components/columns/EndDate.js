// @flow

import * as React from 'react';
import moment from 'moment';
import { useTranslation } from 'react-i18next';

type Props = {
  race: {
    startTime: moment.Moment,
    weekLength: moment.Duration,
  }
};

export default function EndDate({ race }: Props) {
  const { t } = useTranslation();
  const date = moment(race.startTime)
    .local()
    .add(race.weekLength)
    .subtract(1, 'days')
    .toDate();

  return (
    <td>
      <div>
        {t('{{date, YYYY-MM-DD}}', { date })}
      </div>
    </td>
  );
}
