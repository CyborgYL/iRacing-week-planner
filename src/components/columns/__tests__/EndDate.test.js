import moment, { duration } from 'moment';
import React from 'react';
import { shallow } from 'enzyme';
import { describe, test } from '@jest/globals';

import EndDate from '../EndDate';

describe('components/columns/EndDate', () => {
  test('renders correctly', () => {
    const component = shallow(
      <EndDate
        race={{
          startTime: moment('2020-09-22T01:00:00.000Z'),
          weekLength: duration({ days: 7 }),
        }}
      />
    );

    expect(component).toMatchSnapshot();
    expect(component.text()).toBe("2020-09-28");
  });
});
