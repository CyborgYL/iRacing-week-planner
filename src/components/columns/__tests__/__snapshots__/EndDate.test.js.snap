// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/columns/EndDate renders correctly 1`] = `
<td>
  <div>
    2020-09-28
  </div>
</td>
`;
